README.md
# Extension tesseract

This extension is an umbrella for a suite of extensions forming the Tesseract framework.
It provides [a general manual about the framework](https://docs.typo3.org/typo3cms/extensions/tesseract/)
and some basic libraries and classes.

For more general information please check out the
[dedicated web site](http://www.typo3-tesseract.com/).


ChangeLog
2017-06-06 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 2.1.0

2017-06-02 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Verified compatibility with TYPO3 v8

2016-04-15 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Cleaned up extension configuration helper, resolves #75638
	* Cleaned up code, resolves #75639
	* Moved language file to XLIFF, resolves #75640
	* Released to TER as version 2.0.0, resolves #75643

2016-04-13 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Moved documentation to reST, resolves #69243

2016-04-12 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Reformatted code using PSR-2, resolves #75536

2015-08-21 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Moved to namespaces, verified compatibility with TYPO3 CMS 7, resolves #69242

2014-05-09 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Verified compatibility with TYPO3 CMS 6.2, resolves #58683
	* Released to TER as version 1.8.0, resolves #58684

2013-11-06 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Removed old require_once() calls, resolves #53386

2013-06-21 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.7.0

2013-05-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added template retrieval utility (FAL support), resolves #48470

2013-04-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Verified compatibility with TYPO3 CMS 6.1, resolves #47453

2013-04-11 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added API for getting controller properties, resolves #38398

2013-04-01 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Verified compatibility with TYPO3 CMS 6.0, resolves #46842

2012-07-30 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.6.0

2012-07-23 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added new "engine" property to Data Filter structure and base class, resolves #39190

2012-07-10 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.5.0

2012-07-07 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added check for missing service, resolves #38739

2012-06-26 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Cleaned up some code, verified compatibility with TYPO3 4.7, raised requirements to TYPO3 4.5, resolves #38400

2012-04-19 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.4.0

2012-04-03 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Removed obsolete method tx_tesseract_utilities::wrapMessage(), resolves #32280

2012-03-20 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Moved initEmptyDataStrucure() method to data provider base class instead of interface, resolves #35048
	* Corrected creation of empty data structure to take structure type into account, resolves #35051

2012-02-06 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.3.0

2012-01-19 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added documentation about new negate flag for filter conditions, resolves #33322

2012-01-12 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Corrected description of tx_tesseract_datacontroller_output::getPrefixId(), resolves #33154
	* Added API for message queue handling in the controller, resolves #9072

2012-01-08 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Implemented improvements to the API, resolves #33042

2011-12-03 Francois Suter (Cobweb)  <typo3@cobweb.ch>

    * Added utility method for getting a language object, resolves #32281

2011-08-16 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.2.0

2011-07-18 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added API for empty filter check, resolves #28285

2011-07-17 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added documentation about void filters, resolves #28276

2011-05-18 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.1.0

2011-05-17 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Enabled use of TypoScript to define default values for components properties, resolves #13660

2011-03-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Updated documentation for inclusion of filter information in data structure, resolves #13972
	* Updated documentation to describe newer Tesseract components
	* Released to TER as version 1.0.2

2010-12-28 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.0.1

2010-12-20 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Data filter information should contain original filter information, resolves #11599
	* Added warning about extensions installation order, resolves #11549

2010-11-15 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Improved wording of installation check screen, resolves #10162

2010-10-05 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added data setter method to Data Filter interface, resolves #10059

2010-09-27 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* First public release


ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "tesseract".
 *
 * Auto generated 06-06-2017 11:33
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array (
  'title' => 'Project Tesseract',
  'description' => 'This projects covers a suite of extensions designed to facilitate the extraction of data and their display, using controllers and standard data formats. More info on http://www.typo3-tesseract.com/',
  'category' => 'misc',
  'author' => 'Francois Suter (Cobweb)',
  'author_email' => 'typo3@cobweb.ch',
  'state' => 'stable',
  'uploadfolder' => 0,
  'createDirs' => '',
  'clearCacheOnLoad' => 0,
  'author_company' => '',
  'version' => '2.1.0',
  'constraints' => 
  array (
    'depends' => 
    array (
      'php' => '5.5.0-0.0.0',
      'typo3' => '7.6.0-8.99.99',
      'expressions' => '2.0.0-0.0.0',
    ),
    'conflicts' => 
    array (
    ),
    'suggests' => 
    array (
    ),
  ),
  '_md5_values_when_last_written' => 'a:48:{s:9:"ChangeLog";s:4:"4482";s:9:"README.md";s:4:"9add";s:13:"composer.json";s:4:"59d2";s:21:"ext_conf_template.txt";s:4:"3cd0";s:12:"ext_icon.gif";s:4:"37ae";s:14:"ext_tables.php";s:4:"5fa5";s:21:"Classes/Tesseract.php";s:4:"7a78";s:43:"Classes/Component/DataConsumerInterface.php";s:4:"cf7b";s:45:"Classes/Component/DataControllerInterface.php";s:4:"093c";s:51:"Classes/Component/DataControllerOutputInterface.php";s:4:"715f";s:41:"Classes/Component/DataFilterInterface.php";s:4:"2adc";s:43:"Classes/Component/DataProviderInterface.php";s:4:"861f";s:31:"Classes/Exception/Exception.php";s:4:"bb87";s:47:"Classes/Exception/InvalidComponentException.php";s:4:"6956";s:47:"Classes/Exception/MissingComponentException.php";s:4:"0db9";s:42:"Classes/Exception/MissingFileException.php";s:4:"b674";s:43:"Classes/Exception/MissingValueException.php";s:4:"d3c2";s:41:"Classes/Frontend/PluginControllerBase.php";s:4:"8b5b";s:29:"Classes/Service/Component.php";s:4:"b2ad";s:32:"Classes/Service/ConsumerBase.php";s:4:"d99c";s:34:"Classes/Service/ControllerBase.php";s:4:"7553";s:30:"Classes/Service/FilterBase.php";s:4:"1e71";s:40:"Classes/Service/FrontendConsumerBase.php";s:4:"ec43";s:32:"Classes/Service/ProviderBase.php";s:4:"ac37";s:55:"Classes/Utility/ExtensionManagerConfigurationHelper.php";s:4:"2036";s:29:"Classes/Utility/Utilities.php";s:4:"0ca0";s:26:"Documentation/Includes.txt";s:4:"6d5f";s:23:"Documentation/Index.rst";s:4:"27c1";s:26:"Documentation/Settings.yml";s:4:"715c";s:33:"Documentation/AppendixA/Index.rst";s:4:"cf06";s:37:"Documentation/Configuration/Index.rst";s:4:"6c55";s:34:"Documentation/Developers/Index.rst";s:4:"129a";s:48:"Documentation/Developers/BaseComponent/Index.rst";s:4:"6b9b";s:45:"Documentation/Developers/Controller/Index.rst";s:4:"2cfb";s:47:"Documentation/Developers/DataConsumer/Index.rst";s:4:"aa42";s:46:"Documentation/Developers/DataFilters/Index.rst";s:4:"81f6";s:47:"Documentation/Developers/DataProvider/Index.rst";s:4:"3b9c";s:36:"Documentation/Fundamentals/Index.rst";s:4:"dd5a";s:57:"Documentation/Fundamentals/DataFiltersStructure/Index.rst";s:4:"3adb";s:48:"Documentation/Fundamentals/MainConcept/Index.rst";s:4:"945c";s:63:"Documentation/Fundamentals/StandardizedDataStructures/Index.rst";s:4:"41dd";s:47:"Documentation/Images/ExtensionConfiguration.png";s:4:"6eed";s:46:"Documentation/Images/TesseractGlobalSchema.png";s:4:"b73c";s:36:"Documentation/Installation/Index.rst";s:4:"79ab";s:36:"Documentation/Introduction/Index.rst";s:4:"781b";s:37:"Documentation/KnownProblems/Index.rst";s:4:"3ca6";s:33:"Documentation/Resources/Index.rst";s:4:"bf3f";s:40:"Resources/Private/Language/locallang.xlf";s:4:"6c1b";}',
);

